sov-rollup-starter/src/bin/node.rs

```diff
@@ -1,53 +1,59 @@
 //! This binary runs the rollup full node.
 
-use std::env;
 use std::path::PathBuf;
+use std::str::FromStr;
 
 use anyhow::Context;
-use sov_risc0_adapter::host::Risc0Host;
-use sov_rollup_interface::mocks::{MockAddress, MOCK_SEQUENCER_DA_ADDRESS};
-use sov_rollup_starter::da::{start_da_service, DaConfig};
-use sov_rollup_starter::rollup::Rollup;
+use clap::Parser;
+use sov_modules_rollup_template::{Rollup, RollupProverConfig, RollupTemplate};
+use sov_rollup_interface::mocks::MockDaConfig;
+use sov_rollup_starter::rollup::StarterRollup;
 use sov_stf_runner::{from_toml_path, RollupConfig};
-use stf_starter::{get_genesis_config, GenesisPaths};
+use stf_starter::genesis_config::GenesisPaths;
 use tracing::info;
-use tracing_subscriber::filter::LevelFilter;
-use tracing_subscriber::EnvFilter;
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::{fmt, EnvFilter};
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// The path to the rollup config.
+    #[arg(long, default_value = "rollup_config.toml")]
+    rollup_config_path: String,
+}
 
 #[tokio::main]
 async fn main() -> Result<(), anyhow::Error> {
-    // Initialize a logger for the demo
-    let subscriber = tracing_subscriber::fmt()
-        .with_env_filter(
-            EnvFilter::builder()
-                .with_default_directive(LevelFilter::INFO.into()) // If no logging config is set. default to `info` level logs
-                .from_env_lossy(), // Parse the log level from the RUST_LOG env var if set
-        ) // Try to override logging config from RUST_LOG env var
-        .finish();
-    tracing::subscriber::set_global_default(subscriber)
-        .context("Unable to set global default subscriber")?;
-
-    // Read the rollup config from a file
-    let rollup_config_path = env::args()
-        .nth(1)
-        .map(PathBuf::from)
-        .unwrap_or_else(|| PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("rollup_config.toml"));
+    // Initializing logging
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(EnvFilter::from_str("debug,hyper=info").unwrap())
+        .init();
+
+    let args = Args::parse();
+    let rollup_config_path = args.rollup_config_path.as_str();
+
+    let rollup = new_rollup_with_mock_da(
+        &GenesisPaths::from_dir("test-data/genesis/"),
```

Comment:
Looks like this should come from `Args` too, with default `test-data/genesis`

Reply:
done